traffic_ops/testing/api/v5/deliveryservices_keys_test.go

```diff
@@ -333,6 +334,38 @@ func DeliveryServiceSSLKeys(t *testing.T) {
 	if dsSSLKey.Certificate.CSR == "" {
 		t.Errorf("expected a valid CSR, but got nothing")
 	}
+
+	var otherKey *tc.DeliveryServiceSSLKeys
+	for _, ds := range testData.DeliveryServices {
+		if !(*ds.Protocol == tc.DSProtocolHTTPS || *ds.Protocol == tc.DSProtocolHTTPAndHTTPS || *ds.Protocol == tc.DSProtocolHTTPToHTTPS) {
+			continue
+		}
+		var err error
+		dsSSLKey := new(tc.DeliveryServiceSSLKeys)
+		for tries := 0; tries < 5; tries++ {
+			time.Sleep(time.Second)
+			var sslKeysResp tc.DeliveryServiceSSLKeysResponse
+			sslKeysResp, _, err = TOSession.GetDeliveryServiceSSLKeys(*ds.XMLID, client.RequestOptions{})
+			*dsSSLKey = sslKeysResp.Response
+			if err == nil && dsSSLKey != nil {
+				break
+			}
+		}
+		if dsSSLKey != nil {
+			otherKey = dsSSLKey
+			break
+		}
+	}
+	assert.RequireNotNil(t, otherKey, "Expected to find another DS  with a valid SSL certificate")
```

Comment:
`dsSSLKey` is never `nil` at any point, so the check after the inner `for` loop will always be true, and this assertion can never fail. To make it work, I think you'd need to only use `new` when setting the value of `dsSSLKey` in that inner loop, and just declare it as a nil pointer above that. I'm sure the test would fail anyway because parsing a blank cert should probably return an error, but as-is this is an unused check.